encoding/protobuf.go

```diff
@@ -0,0 +1,98 @@
+package encoding
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/jhump/protoreflect/desc"
+	"github.com/jhump/protoreflect/dynamic"
+	"github.com/yarpc/yab/transport"
+)
+
+// SplitMethod takes a method name like Service::Method and splits it
+// into Service and Method.
+func SplitMethod(fullMethod string) (svc, method string, err error) {
+	parts := strings.Split(fullMethod, "::")
+	switch len(parts) {
+	case 1:
+		return parts[0], "", nil
+	case 2:
+		return parts[0], parts[1], nil
+	default:
+		return "", "", fmt.Errorf("invalid proto method %q, expected form package.Service::Method", fullMethod)
+	}
+}
+
+type protoSerializer struct {
+	fullMethodName string
+	method         *desc.MethodDescriptor
+}
+
+// NewProtobuf returns a protobuf serializer.
+func NewProtobuf(fullMethodName string, source ProtoDescriptorSource) (Serializer, error) {
+	svc, mth, err := SplitMethod(fullMethodName)
+	if err != nil {
+		return nil, err
+	}
+	dsc, err := source.FindSymbol(svc)
+	if err != nil {
+		return nil, fmt.Errorf("failed to query for service for symbol %q: %v", svc, err)
+	}
+	sd, ok := dsc.(*desc.ServiceDescriptor)
+	if !ok {
+		return nil, fmt.Errorf("target server does not expose service %q", svc)
+	}
+	mtd := sd.FindMethodByName(mth)
+	if mtd == nil {
+		return nil, fmt.Errorf("service %q does not include a method named %q", svc, mth)
+	}
+	return protoSerializer{
```

Comment:
I'd make this a pointer (and attach all methods below to `*protoSerializer`, I'm betting @akshayjshah will have input re: the memory argument however

Reply:
This isn't that performance-sensitive, so it doesn't matter here. Unless you're being *very* picky about performance or specifically want pass-by-value semantics, pointers are more common in Go.